mhkit/wave/performance.py

```diff
@@ -187,79 +190,177 @@ def power_matrix(LM, JM):
+        water depth will be set to deep. Default ratio = 2.
+    show_values : bool (optional)
+        Show values on the scatter diagram. Default = False.
+    savepath: string (optional)
+        Path to save figure. Terminate with '\'. Default="".
+
+    Returns
+    ---------
+    LM: xarray dataset
+        Capture length matrices
+
+    maep_matrix: float
+        Mean annual energy production
+    """
+    assert isinstance(S, (pd.DataFrame,pd.Series)), 'S must be of type pd.DataFrame or pd.Series'
+    assert isinstance(h, (int,float)), 'h must be of type int or float'
+    assert isinstance(P, (np.ndarray, pd.Series)), 'P must be of type np.ndarray or pd.Series'
+    assert isinstance(deep, bool), 'deep must be of type bool'
+    assert isinstance(rho, (int,float)), 'rho must be of type int or float'
+    assert isinstance(g, (int,float)), 'g must be of type int or float'
+    assert isinstance(ratio, (int,float)), 'ratio must be of type int or float'
+
+    # Compute the enegy periods from the spectra data
+    Te = wave.resource.energy_period(S, frequency_bins=frequency_bins)
+    Te = Te['Te']
+
+    # Compute the significant wave height from the NDBC spectra data
+    Hm0 = wave.resource.significant_wave_height(S, frequency_bins=frequency_bins)
+    Hm0 = Hm0['Hm0']
+
+    # Compute the energy flux from spectra data and water depth
+    J = wave.resource.energy_flux(S, h, deep=deep, rho=rho, g=g, ratio=ratio)
+    J = J['J']
+
+    # Calculate capture length from power and energy flux
+    L = wave.performance.capture_length(P,J)
+
+    # Generate bins for Hm0 and Te, input format (start, stop, step_size)
+    Hm0_bins = np.arange(0, Hm0.values.max() + .5, .5)
+    Te_bins = np.arange(0, Te.values.max() + 1, 1)
+
+    # Create capture length matrices for each statistic based on IEC/TS 62600-100
+    # Median, sum, frequency additionally provided
+    LM = xr.Dataset()
+    LM['mean'] = wave.performance.capture_length_matrix(Hm0, Te, L, 'mean', Hm0_bins, Te_bins)
+    LM['std'] = wave.performance.capture_length_matrix(Hm0, Te, L, 'std', Hm0_bins, Te_bins)
+    LM['median'] = wave.performance.capture_length_matrix(Hm0, Te, L, 'median', Hm0_bins, Te_bins)
+    LM['count'] = wave.performance.capture_length_matrix(Hm0, Te, L, 'count', Hm0_bins, Te_bins)
+    LM['sum'] = wave.performance.capture_length_matrix(Hm0, Te, L, 'sum', Hm0_bins, Te_bins)
+    LM['min'] = wave.performance.capture_length_matrix(Hm0, Te, L, 'min', Hm0_bins, Te_bins)
+    LM['max'] = wave.performance.capture_length_matrix(Hm0, Te, L, 'max', Hm0_bins, Te_bins)
+    LM['freq'] = wave.performance.capture_length_matrix(Hm0, Te, L,'frequency', Hm0_bins, Te_bins)
 
+    # Create wave energy flux matrix using mean
+    JM = wave.performance.wave_energy_flux_matrix(Hm0, Te, J, 'mean', Hm0_bins, Te_bins)
 
+    # Calculate maep from matrix
+    maep_matrix = wave.performance.mean_annual_energy_production_matrix(LM['mean'].to_pandas(), JM, LM['freq'].to_pandas())
```

Comment:
Must use LM['mean'] indexing instead LM.mean as .mean is a method of xarray

Must use to_pandas() not to_dataframe() method as former inputs data correctly as 2d matrix